### DISCUSSION TITLE
Rendering invisible or low-opacity objects

### DISCUSSION THREAD

**Discussion by SimpleGeometry** [google/brax#545] [General]
I'm looking to make invisible objects with rgba="0 0 0 0" (alpha = 0 should produce an invisible object in Mujoco I believe), though brax's render function does not respond to the alpha field (e.g. should be half-transparent for alpha=0.5 or invisible for alpha=0).
- I'm using brax.io.html.render, which I think is calling https://github.com/google/brax/blob/main/brax/visualizer/js/system.js#L165. 

I'm sorry about making all these feature requests, so this is my best effort to look into a potential implementation, in case it saves the devs some time (have not looked into how to test changes to the repo, but I'd love to make a PR instead if I knew how to test):
- For responding to alpha: it seems like three.js's MeshStandardMaterial and MeshPhongMaterial both have .alphaMap properties. Or as materials they have .opacity and .transparent fields (needs to set transparent=True I think).
- For becoming invisible when alpha = 0: .opacity = 0 and .transparent = True.
- Maybe the fix to both of these is to set .transparent = True, and then .opacity = rgba[3]?
- My references: https://threejs.org/docs/#api/en/materials/MeshPhongMaterial, https://threejs.org/docs/#api/en/materials/MeshStandardMaterial, https://threejs.org/docs/#api/en/materials/Material.transparent, https://github.com/mrdoob/three.js/issues/22598

An alternative question: the reason I'm using invisible walls is to constrain the motion of a freejoint cube. Is there any other way in MJX to constrain the range of a freejoint?
- One possible approach is to manually add 3 slide joints and a ball joint, then add a range to the slide joints, but I don't see why MJX constrains freejoints to not allow range inputs.

**Comment by btaba** [comment]
Hi @SimpleGeometry , adding 3 slide joints and a ball joint makes sense to me to constrain the slide joint motion. Free joints don't have joint limits because they are free joints! 
As far for the renderer, feel free to open up a PR once you've modified the viewer code and tested it locally (please include screenshots)!
